lib/ModelMixins/GltfMixin.ts

```diff
@@ -25,24 +25,54 @@ const Axis: Axis = require("terriajs-cesium/Source/Scene/Axis").default;
 
 type GltfModel = Model<GltfTraits>;
 
+export interface GltfTransformationJson {
+  origin: {
+    latitude?: number;
+    longitude?: number;
+    height?: number;
+  };
+  rotation: {
+    heading?: number;
+    pitch?: number;
+    roll?: number;
+  };
+  scale?: number;
+}
+
 function GltfMixin<T extends Constructor<GltfModel>>(Base: T) {
   abstract class GltfMixin extends ShadowMixin(
     CatalogMemberMixin(MappableMixin(Base))
   ) {
+    // Create stable instances of DataSource and Entity instead
+    // of generating a new one each time the traits change and mobx recomputes.
+    // This vastly improves the performance.
+    private readonly _dataSource = new CustomDataSource("glTF Model");
```

Comment:
Maybe add a comment that `_dataSource` and `_modelEntity` should not be modified outside of their computeds, `private` is nice but could still modify properties...